sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -341,17 +341,20 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     throw new ParseException("INSERT OVERWRITE DIRECTORY is not supported", ctx)
   }
 
+  // We do not allow columns aliases in some cases.
+  private def notAllowColumnAliases(ctx: TableAliasContext, msg: String): Unit = {
+    if (ctx.identifierList() != null) {
+      throw new ParseException(msg, ctx.identifierList())
+    }
+  }
+
   override def visitDeleteFromTable(
       ctx: DeleteFromTableContext): LogicalPlan = withOrigin(ctx) {
 
     val tableId = visitMultipartIdentifier(ctx.multipartIdentifier)
     val tableAlias = if (ctx.tableAlias() != null) {
       val ident = ctx.tableAlias().strictIdentifier()
```

Comment:
we can do further
```
private def getTableAliasWithoutColumnAlias ... = {
  val ident = ctx.strictIdentifier()
  if ...
  if (ident != null) Some(ident.getText) else None
}
```

then here just call `getTableAliasWithoutColumnAlias(ctx.tableAlias())`

Reply:
done.